File: cartipay-main/Pardakht.PardakhtPay/FrontEnd/Pardakht.PardakhtPay.AdminPortal/src/app/models/card-to-card-account.ts
import { GenericHelper } from "../helpers/generic";

export class CardToCardAccount {
    id: number;
    bankLoginId: number;
    bankAccountId: number;
    loginGuid: string;
    accountGuid: string;
    cardNumber: string;
    cardHolderName: string;
    safeAccountNumber: string;
    transferThreshold: number;
    isActive: boolean;
    isTransferThresholdActive: boolean;
    transferThresholdLimit: number;
    tenantGuid: string;
    ownerGuid: string;
    friendlyName: string;
    loginType: number;
    accountNo: string;
    switchOnLimit: boolean;
    switchIfHasReserveAccount: boolean;
    switchLimitAmount: number;
    accountType: number;
    switchCreditDailyLimit: number;

    constructor(data?: any) {
        GenericHelper.populateData(this, data);
    }
}

export enum AccountType{
    None = 0,
    Withdrawal = 1,
    Deposit = 2,
    Both = 3
}

File: cartipay-main/Pardakht.PardakhtPay/BackEnd/Pardakht.PardakhtPay.Shared.Models/Entities/CardToCardAccount.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Pardakht.PardakhtPay.Shared.Models.Models;

namespace Pardakht.PardakhtPay.Shared.Models.Entities
{
    public class CardToCardAccount : BaseEntity, ITenantGuid, IOwnerGuid
    {
        [Required]
        public string LoginGuid { get; set; }

        [Required]
        public string AccountGuid { get; set; }

        //[Required]
        public string CardNumber { get; set; }

        [Required]
        public string CardHolderName { get; set; }

        //[Required]
        public string SafeAccountNumber { get; set; }

        [Column(TypeName = "decimal(18, 2)")]
        public decimal TransferThreshold { get; set; }

        public bool IsActive { get; set; }

        public bool IsTransferThresholdActive { get; set; }

        [Column(TypeName = "decimal(18, 2)")]
        public decimal? TransferThresholdLimit { get; set; }

        public string TenantGuid { get; set; }

        public string OwnerGuid { get; set; }

        public int LoginType { get; set; }

        public bool SwitchOnLimit { get; set; }

        public bool SwitchIfHasReserveAccount { get; set; }

        [Column(TypeName = "decimal(18, 2)")]
        public decimal? SwitchLimitAmount { get; set; }

        [Column(TypeName = "decimal(18, 2)")]
        public decimal? SwitchCreditDailyLimit { get; set; }
    }
}
